### ISSUE TITLE
FR: touch gestures support in core?

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#11784]
I was asked if I could add touch gestures in Tidypics for navigating to prev/next pictures. It was also suggested that it would be even better if touch gestures would also work for all Elgg plugins (e.g. loading prev/next blog entry by touch gestures). I think it would make sense to be able to navigate in lists by panning left/right and browsing to the prev/next entries in full view with touch gestures on mobile devices. In general, it would be easier to pan than to hit the small navigation errors on small smartphone screen.

With the "mobile first" approach of Elgg 3.0 I wonder if the lack of support of touch gestures in core is the one detail missing. Even if the touch gestures wouldn't be made use of by core plugins yet, wouldn't it make sense to add some basic support (i.e. include a library that provides touch gestures support)? With such a library in Elgg core it wouldn't be necessary for each plugin that would support touch gestures to include a library on its own (resulting in duplication and likely also fragmentation with different libs used).

**Comment by hypeJunction** [comment]
I am not opposed. Might be a good addition if the API is done right. On the other hand I think most touch events are supported natively by JavaScript, so you can just listen to swipe events. https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Using_Touch_Events

**Comment by jdalsem** [comment]
@iionly do you have any idea which kind of library could help us to add these features?

**Comment by iionly** [comment]
I came across http://hammerjs.github.io/ but I had no time for testing it out yet. I can't say if it's of use or if there might be some other, better library available. But it supports swipeleft, swiperight at least which might be the most useful. While there's also a jquery plugin wrapper for hammerjs available I think the hammerjs lib is not available on packagist.

For simple touch events it might not be necessary to add a library. But it might be useful for touch gestures support to use a library (regarding browser support).

**Comment by iionly** [comment]
I tried out the hammer.js script and created a plugin with it: https://github.com/iionly/hammertime. Currently it supports swipe left/right on lists with pagination and on Tidypics full image view pages. If anyone would like to give it a try I would appreciate any feedback (especially if testing on any mobile devices as I've only tested with touch emulation in desktop browsers...). Not sure anymore about adding support for gestures in core at the moment as I'm unsure about what other use cases would gain from gestures support.
